galley/pkg/config/processing/snapshotter/analyzingdistributor.go

```diff
@@ -206,6 +208,16 @@ FilterMessages:
 			}
 		}
 
+		// Filter out any messages on resources with suppression annotations.
+		if m.Resource != nil && m.Resource.Metadata.Annotations[annotation.GalleyAnalyzeSuppress.Name] != "" {
+			for _, code := range strings.Split(m.Resource.Metadata.Annotations[annotation.GalleyAnalyzeSuppress.Name], ",") {
+				if code == "*" || m.Type.Code() == code {
+					continue FilterMessages
+				}
+			}
+
+		}
```

Comment:
1. Add a debug-level message here, like we did for the other suppression filtering below. 

2. You may have already considered this, but what do you think about using annotations to generate `AnalysisSuppression` objects that we simply add to the `suppressions` list we're already processing below? It avoids some duplicate logic.

Reply:
1. Good call. added.

2. That would also work. It might be potentially confusing though during debugging why something was suppressed as it's harder to figure out whether it was due to an annotation or a command-line argument. You could fix this by adding a tag to the suppression itself indicating its source... I suspect this will be more code than the reduction in similar logic.